### PR TITLE
feat: only and skip support

### LINKED ISSUE #5: Rhum.only()
**What**

Develop the following method:

* `Rhum.only`

Method should be public and should tell Rhum to only run that test plan/suite/case **under** the parent. For example:

* If 1 suite of 2 is set to only under a plan, then the suite that doesnt have only should be skipped

**Why**
This is a testing framework dammit!

**Checklist**

* Update README where needed

* Unit & integration tests

**Resources**

Use the following [issue](https://github.com/denoland/deno/issues/4092) as a guideline if possible

**Pseudo-code for Implementation**

_All we have for information at the moment_

Needs to account for plan/suite/case, and Rhum/RhumRunner should know that there is only one to test

```typescript
Rhum.only: function (name: string, fn: Function) {
}
```

### PR DESCRIPTION
**Pull request by ebebbington**
Fixes issue 5

**WIP**

**Developments so far**

* Implemented only usage
    * Only a single suite or case can be only'ed, if more than 1 is, theen rhum throws an error
    * Will show all other tests as ignored
* added skip support
    * same as above but as many test *'s can be skipped
* Temporarily changed the output of rhum, see below

**What's left**

* Complete and fix integration tests for only
* add unit and integration tests for skip
* add unit tests for only method
* correct the output format, and adjust tests for this

**Different types of output**

image

image

### CODE REVIEW

File: mod.ts [hunk 1]
```diff
@@ -224,9 +224,29 @@ export class RhumRunner {
...
+        name,
+        new_name: this.formatTestCaseName(name),
+        testFn: cb,
+        only: true
+      });
+    } else if (this.passed_in_test_plan) { // is a test suite being skipped
+      this.passed_in_test_suite = name;
+      this.plan.suites![name] = {cases: [], only: true};
+      cb();
+    }
```
Comment (ebebbington) [inline mod.ts:247]: Need to figure this out

**Comment by ebebbington** [comment]
Developments so far:

* Implemented `only` usage
    * Only a single suite or case can be only'ed, if more than 1 is, theen rhum throws an error
    * Will show all. other tests as ignored
* Temporarily changed the output of rhum which hasn't been best received, so it needs to be changed again - i'm proposing we ditch the current format but keep the `test ` bit in, so it's easier to write tests for

**Comment by ebebbington** [comment]
Decided to add skip support as well, because the only way only or skip will land, is if we ditch current output format, so may as well do it in one pr

**Comment by crookse** [comment]
can we rework this PR to be pointed to v2? or should we trash it and start over with a new branch, but taking the logic from this branch as a starting point?

**Comment by ebebbington** [comment]
Oh we can trash this PR, I was only keeping it open as a reference really until we landed your pr, if this PR is no use to you then feel free to close it and delete the branch 👍

**Comment by crookse** [comment]
i think it's still useful. we can keep it open as a reference for .only(). i like the edge cases so let's keep it open for now

**Comment by ebebbington** [comment]
Alright

**Comment by MVEMCJSUNPE** [comment]
Do we need only and skip properties for every test interface? It seems like it just overcomplicates things. For example, what happens if both only and skip are set to true?

**Comment by ebebbington** [comment]
Hey, yes because it means any test suite or case can be skipped, and i believe i added checks for if both are used but i can't remember, this PR was from a while ago :P Tbh, we're only keeping this PR up for reference purposes for the rhum v2 PR, none of this code will end up in master and when rhum v2 is in master, this pr will be closed and the code lost forever 😢

**Comment by MVEMCJSUNPE** [comment]
I know. I just realized that this getting this to work is probably not possible if we keep using deno.std/testing as a base, unless we find some way to change the message format for deno.std/testing. The only way I can see to get this work is just to write our own base for testing instead of working off of deno.std/testing.

**Comment by MVEMCJSUNPE** [comment]
So basically we would have to write our own "deno:cli/rt/40_testing.js".

**Comment by crookse** [comment]
it might be that we don't need this feature now that i think about it. i say this because rhum v2 CLI has the --filter-test-case and --filter-test-suite options

**Comment by MVEMCJSUNPE** [comment]
I think those options achieve the same functionality as skip, but I'm not sure those options achieve the same functionality as only. I'm pretty sure ebebbington's code would have worked if Deno's testing api had more customizable format options.

**Comment by ebebbington** [comment]
My pr was going to use a different test definition so we could use skip and only, but this meant the output would have changed in the shell for users

Where as now with Rhum v2, we have good output and can still ignore and only tests

**Comment by ebebbington** [comment]
For reference, this PR is only open to aid in eric's v2 PR, should any of the work i have done come in handy.

Once the v2 pr is merged, this will be closed **without** merging

**Comment by crookse** [comment]
not doing this anymore. v2 will be a test double framework
